csrc/includes/cpu_adam.h

```diff
@@ -95,20 +95,44 @@ class Adam_Optimizer {
         _betta1_t *= _betta1;
         _betta2_t *= _betta2;
         if (_step < step) {
-            printf("Warning: updating optimizer's state to match the step count at %d\n", step);
-            while (_step < step) {
-                _step++;
-                _betta1_t *= _betta1;
-                _betta2_t *= _betta2;
-            }
+            printf("Warning: updating optimizer's state to match the step count at %lu\n", step);
+            _betta1_t = std::pow(_betta1, step);

```

Comment:
I think in that case, the _step has become zero (https://github.com/microsoft/DeepSpeed/blob/059f6aab5bfe2abb72106af29ced1b8575ff4fd9/csrc/includes/cpu_adam.h#L62), since this means a new training process. Is in it right?

Reply:
When loading a new checkpoint, the optimizer will initialize, am I right?